spi/common/boot-spi/src/main/java/org/eclipse/edc/spi/result/Result.java

```diff
@@ -41,6 +41,10 @@ public static <T> Result<T> success(@NotNull T content) {
         return new Result<>(content, null);
     }
 
+    public static <T> Result<T> successNullable(@Nullable T content) {
```

Comment:
why not remove the `@NotNull` annotation from the `success` method? I think it's perfectly fine to have a null content (e.g. for `Void` type)

Reply:
that was the alternative approach. My thinking was to make it more explicit, since returning `null` should be more of an edge case.
I was kinda in-between anyway, so i'll update the code.

Reply:
In any case on the "client" side there will no way to know if this edge case was put in place other than checking if the content is null or not. IIRC at first it was put "not nullable" because the logic was: "if content is null result is failed", but it ended with "if failure is not null result is failed"

Reply:
yep, makes sense. users of a method that returns a `Result` would be well-advised to read javadoc anyway.